recipes/libzippp/all/conanfile.py

```diff
@@ -1,88 +1,108 @@
     options = {
         "shared": [True, False],
         "fPIC": [True, False],
-        "with_encryption": [True, False]
+        "with_encryption": [True, False],
     }
     default_options = {
         "shared": False,
         "fPIC": True,
-        "with_encryption": False
+        "with_encryption": False,
     }
-    _cmake = None
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _min_cppstd(self):
+        return 11
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
 
-    def validate(self):
-        libzippp_version = str(self.version)
-        if libzippp_version != "4.0" and len(libzippp_version.split("-")) != 2:
-            raise tools.ConanInvalidConfiguration("{}: version number must include '-'. (ex. '5.0-1.8.0')".format(self.name))
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("zlib/1.2.11")
-        if tools.Version(self.version) == "4.0":
+        self.requires("zlib/1.2.13")
+        if Version(self.version) == "4.0":
             self.requires("libzip/1.7.3")
         else:
-            libzip_version = str(self.version).split("-")[1]
-            self.requires("libzip/{}".format(libzip_version))
+            versions = str(self.version).split("-")
+            if len(versions) == 2:
+                self.requires(f"libzip/{versions[1]}")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+
+        libzippp_version = str(self.version)
+        if libzippp_version != "4.0" and len(libzippp_version.split("-")) != 2:
+            raise ConanInvalidConfiguration(f"{self.ref}: version number must include '-'. (ex. '5.0-1.8.0')")
+
+        if self.settings.compiler == "clang" and self.settings.compiler.get_safe("libcxx") == "libc++":
+            raise ConanInvalidConfiguration(f"{self.ref} does not support clang with libc++. Use libstdc++ instead.")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-            destination=self._source_subfolder, strip_root=True)
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["LIBZIPPP_INSTALL"] = True
-        self._cmake.definitions["LIBZIPPP_INSTALL_HEADERS"] = True
-        self._cmake.definitions["LIBZIPPP_ENABLE_ENCRYPTION"] = self.options.with_encryption
-        self._cmake.configure()
-        return self._cmake
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["CMAKE_CXX_STANDARD"] = 11
```

Comment:
Usually we set the compile min and/or do an `if not` the setting is defined for the 1.x case 🤔